superset/explore/api.py

```diff
@@ -0,0 +1,127 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from abc import ABC
+
+from flask import g, request, Response
+from flask_appbuilder.api import BaseApi, expose, protect, safe
+
+from superset.charts.commands.exceptions import ChartNotFoundError
+from superset.constants import MODEL_API_RW_METHOD_PERMISSION_MAP, RouteMethod
+from superset.explore.commands.get import GetExploreCommand
+from superset.explore.commands.parameters import CommandParameters
+from superset.explore.exceptions import DatasetAccessDeniedError, WrongEndpointError
+from superset.explore.permalink.exceptions import ExplorePermalinkGetFailedError
+from superset.explore.schemas import ExploreContextSchema
+from superset.extensions import event_logger
+from superset.temporary_cache.commands.exceptions import (
+    TemporaryCacheAccessDeniedError,
+    TemporaryCacheResourceNotFoundError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ExploreRestApi(BaseApi, ABC):
```

Comment:
Why abstract class here? I don't see any classes that inherit from this abstract class.